src/vif.jl

```diff
@@ -44,10 +47,11 @@ function StatsAPI.vif(model::RegressionModel)
         throw(ArgumentError("VIF not meaningful for models with only one non-intercept term"))
     # NB: The correlation matrix is positive definite and hence invertible
     #     unless there is perfect rank deficiency, hence the warning in the docstring.
-    return diag(inv(m))
+    # there's no `inv(::SubArray)` method before 1.7
+    # but this matrix is indeed symmetric and the Symmetric wrapper is enough to make this work
+    return diag(inv(Symmetric(m)))
```

Comment:
Is there any performance advantage in using `Symmetric`? Otherwise I'd also use a `VERSION` check to avoid using more complex code on recent Julia versions.

Reply:
There should be -- the inverse of a symmetric matrix is itself symmetric, so you can take advantage of that to get better performance. (In other words, if LinearAlgebra doesn't take advantage of this, it should and may in the future.)

Reply:
OK. In that case let's just drop the comment which seems to imply we only do this as a workaround:
```suggestion
    return diag(inv(Symmetric(m)))
```